src/braket/parametric/free_parameter.py

```diff
@@ -66,7 +66,7 @@ def subs(self, parameter_values: dict[str, Number]) -> Union[FreeParameter, Numb
             Union[FreeParameter, Number]: The substituted value if this parameter is in
             parameter_values, otherwise returns self
         """
-        return parameter_values[self.name] if self.name in parameter_values else self
+        return parameter_values.get(self.name, self)
```

Comment:
Can we check that this does not cause an error if the name is not in `parameter_values`